src/org/FatimaHasT/qamp/zadaci/zadaca7/task2/ElectricCar.java

```diff
@@ -0,0 +1,32 @@
+package org.FatimaHasT.qamp.zadaci.zadaca7.task2;
+
+public class ElectricCar extends Vehicle {
+    private final ElectricEngine engine;
+
+    public ElectricCar(final String model, final double capacity) {
+        super(model);
+        this.engine = new ElectricEngine(capacity);
+    }
+
+    public void accelerate(final double speed) {
+        final double factor = 0.4;
+        final double fuelConsumption = factor * speed;
+        if (this.engine.getCapacity() < fuelConsumption) {
```

Comment:
you could have gotten around checking this in the ElectricCar class, by having a canAccelerate() method in the ElectricEngine class which would perform this check.

Also this would avoid the need for a capacity getter.